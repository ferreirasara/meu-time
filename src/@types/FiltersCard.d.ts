export type FiltersCardProps = {
  selectedCountry: string
  handleSelectCountry: (country: string) => void
  selectedLeague: number
  handleSelectLeague: (league: number) => void
  selectedSeason: number
  handleSelectSeason: (season: number) => void
  selectedTeam: number
  handleSelectTeam: (team: number) => void
}