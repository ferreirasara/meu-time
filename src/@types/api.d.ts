export type Team = {
  team: {
    id: number,
    name: string,
    code: string,
    country: string,
    founded: number,
    national: boolean,
    logo: string,
  },
  venue: {
    id: number,
    name: string,
    address: string,
    city: string,
    capacity: number,
    surface: string,
    image: string,
  },
}

export type Country = {
  name: string,
  code?: string,
  flag?: string,
}

export type League = {
  country: Country,
  league: {
    id: number,
    name: string,
    type: string,
    logo: string,
  }
  seasons: {
    year: number,
    start: string,
    end: string,
    current: boolean,
    coverage: {
      fixtures: {
        events: boolean,
        lineups: boolean,
        statistics_fixtures: boolean,
        statistics_players: boolean
      },
      standings: boolean,
      players: boolean,
      top_scorers: boolean,
      top_assists: boolean,
      top_cards: boolean,
      injuries: boolean,
      predictions: boolean,
      odds: boolean
    }
  }[],
}