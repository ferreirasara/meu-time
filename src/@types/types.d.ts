export type BaseData = {
  lastUpdated?: Date,
  countries?: Country[],
  leagues?: League[],
  seasons?: number[],
}

export type UserData = {
  email: string,
  firstname: string,
  lastname: string,
}

export type RequestData = {
  current: number,
  limitDay: number,
}

export type PlayersTableData = {
  id: number,
  name: string,
  age: number,
  nationality: string,
  photo: string,
}

export type PlayersTableProps = {
  selectedLeague: number
  selectedSeason: number
  selectedTeam: number
  handleError: (message: string) => void
}

export type GoalsChartProps = {
  data?: GeneralStatsData,
  loading: boolean,
}

export type GeneralStatsData = {
  totalDraws: number,
  totalPlayed: number,
  totalWins: number,
  totalLoses: number,
  mostUsedFormation: {
    formation: string,
    played: number,
  },
  goals: {
    against: {
      minute: {
        "0-15": { total: number, percentage: string },
        "16-30": { total: number, percentage: string },
        "31-45": { total: number, percentage: string },
        "46-60": { total: number, percentage: string },
        "61-75": { total: number, percentage: string },
        "76-90": { total: number, percentage: string },
        "91-105": { total: number, percentage: string },
        "106-120": { total: number, percentage: string }
      },
    },
    for: {
      minute: {
        "0-15": { total: number, percentage: string },
        "16-30": { total: number, percentage: string },
        "31-45": { total: number, percentage: string },
        "46-60": { total: number, percentage: string },
        "61-75": { total: number, percentage: string },
        "76-90": { total: number, percentage: string },
        "91-105": { total: number, percentage: string },
        "106-120": { total: number, percentage: string }
      },
    },
  }
}

export type GeneralStatsProps = {
  selectedLeague: number
  selectedSeason: number
  selectedTeam: number
  handleError: (message: string) => void
}

export type FiltersCardProps = {
  selectedCountry: string
  handleSelectCountry: (country: string) => void
  selectedLeague: number
  handleSelectLeague: (league: number) => void
  selectedSeason: number
  handleSelectSeason: (season: number) => void
  selectedTeam: number
  handleSelectTeam: (team: number) => void
  localStorageDataLoading: boolean
  handleError: (message: string) => void
}

export type Team = {
  team: {
    id: number,
    name: string,
    code: string,
    country: string,
    founded: number,
    national: boolean,
    logo: string,
  },
  venue: {
    id: number,
    name: string,
    address: string,
    city: string,
    capacity: number,
    surface: string,
    image: string,
  },
}

export type Country = {
  name: string,
  code?: string,
  flag?: string,
}

export type League = {
  country: Country,
  league: {
    id: number,
    name: string,
    type: string,
    logo: string,
  }
  seasons: {
    year: number,
    start: string,
    end: string,
    current: boolean,
    coverage: {
      fixtures: {
        events: boolean,
        lineups: boolean,
        statistics_fixtures: boolean,
        statistics_players: boolean
      },
      standings: boolean,
      players: boolean,
      top_scorers: boolean,
      top_assists: boolean,
      top_cards: boolean,
      injuries: boolean,
      predictions: boolean,
      odds: boolean
    }
  }[],
}

export type Player = {
  player: {
    id: number,
    name: string,
    firstname: string,
    lastname: string,
    age: number,
    birth: {
      date: string,
      place: string,
      country: string,
    },
    nationality: string,
    height: string,
    weight: string,
    injured: boolean,
    photo: string,
  },
  // statistics: {
  //   "team": {
  //     "id": 147,
  //     "name": "Coritiba",
  //     "logo": "https://media-1.api-sports.io/football/teams/147.png"
  //   },
  //   "league": {
  //     "id": 71,
  //     "name": "Serie A",
  //     "country": "Brazil",
  //     "logo": "https://media-2.api-sports.io/football/leagues/71.png",
  //     "flag": "https://media-3.api-sports.io/flags/br.svg",
  //     "season": 2016
  //   },
  //   "games": {
  //     "appearences": 6,
  //     "lineups": 6,
  //     "minutes": 438,
  //     "number": null,
  //     "position": "Defender",
  //     "rating": null,
  //     "captain": false
  //   },
  //   "substitutes": {
  //     "in": 0,
  //     "out": 3,
  //     "bench": 0
  //   },
  //   "shots": {
  //     "total": null,
  //     "on": null
  //   },
  //   "goals": {
  //     "total": 0,
  //     "conceded": null,
  //     "assists": null,
  //     "saves": null
  //   },
  //   "passes": {
  //     "total": null,
  //     "key": null,
  //     "accuracy": null
  //   },
  //   "tackles": {
  //     "total": null,
  //     "blocks": null,
  //     "interceptions": null
  //   },
  //   "duels": {
  //     "total": null,
  //     "won": null
  //   },
  //   "dribbles": {
  //     "attempts": null,
  //     "success": null,
  //     "past": null
  //   },
  //   "fouls": {
  //     "drawn": null,
  //     "committed": null
  //   },
  //   "cards": {
  //     "yellow": 0,
  //     "yellowred": 0,
  //     "red": 0
  //   },
  //   "penalty": {
  //     "won": null,
  //     "commited": null,
  //     "scored": null,
  //     "missed": null,
  //     "saved": null
  //   }
  // }[]
}

export type TeamStats = {
  lineups: {
    formation: string,
    played: number
  }[],
  fixtures: {
    draws: { home: number, away: number, total: number }
    loses: { home: number, away: number, total: number },
    played: { home: number, away: number, total: number },
    wins: { home: number, away: number, total: number },
  },
  goals: {
    against: {
      minute: {
        "0-15": { total: number, percentage: string },
        "16-30": { total: number, percentage: string },
        "31-45": { total: number, percentage: string },
        "46-60": { total: number, percentage: string },
        "61-75": { total: number, percentage: string },
        "76-90": { total: number, percentage: string },
        "91-105": { total: number, percentage: string },
        "106-120": { total: number, percentage: string }
      },
    },
    for: {
      minute: {
        "0-15": { total: number, percentage: string },
        "16-30": { total: number, percentage: string },
        "31-45": { total: number, percentage: string },
        "46-60": { total: number, percentage: string },
        "61-75": { total: number, percentage: string },
        "76-90": { total: number, percentage: string },
        "91-105": { total: number, percentage: string },
        "106-120": { total: number, percentage: string }
      },
    },
  }
}

export type InfoModalProps = {
  infoModalOpen: boolean
  onClose: () => void
}