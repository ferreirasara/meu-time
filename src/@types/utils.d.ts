export type BaseData = {
  lastUpdated?: Date,
  countries?: Country[],
  leagues?: League[],
  seasons?: number[],
}