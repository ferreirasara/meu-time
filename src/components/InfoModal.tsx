import { Col, Divider, List, Modal, Row, Statistic, Typography } from "antd"
import { InfoModalProps, RequestData, UserData } from "../@types/types"
import { useState } from "react";

export const InfoModal = ({ infoModalOpen, onClose }: InfoModalProps) => {
  const [showBlur, setShowBlur] = useState<boolean>(true);
  const userDataString = localStorage.getItem('user-data');
  const requestsDataString = localStorage.getItem('requests-data');
  const userDataJson: UserData = JSON.parse(userDataString || "{}");
  const requestsDataJson: RequestData = JSON.parse(requestsDataString || "{}");
  const apiKey = localStorage.getItem('api-key')

  return <Modal
    open={infoModalOpen}
    onCancel={onClose}
    footer={null}
    title="Informações"
  >
    <div style={{
      display: 'flex',
      flexDirection: 'column',
      gap: '16px'
    }}>
      <Row gutter={[16, 16]}>
        <Col span={12}>
          <Statistic title="Requisições feitas" value={requestsDataJson?.current} />
        </Col>
        <Col span={12}>
          <Statistic title="Requisições disponíveis" value={requestsDataJson?.limitDay - requestsDataJson?.current} />
        </Col>
      </Row>
      <Row gutter={[16, 16]}>
        <Col flex='auto'>
          <Divider>Usuário</Divider>
          <List>
            <List.Item>
              <Typography.Text strong>Nome</Typography.Text>: {userDataJson?.firstname} {userDataJson?.lastname}
            </List.Item>
            <List.Item>
              <Typography.Text strong>Email</Typography.Text>: {userDataJson?.email}
            </List.Item>
            <List.Item>
              <Typography.Text strong>API Key</Typography.Text>: <Typography.Text
                style={{ filter: `blur(${showBlur ? '0.15rem' : '0rem'})` }}
                onMouseEnter={() => setShowBlur(false)}
                onMouseLeave={() => setShowBlur(true)}
              >
                {apiKey}
              </Typography.Text>
            </List.Item>
          </List>
        </Col>
      </Row>
    </div>
  </Modal>
}