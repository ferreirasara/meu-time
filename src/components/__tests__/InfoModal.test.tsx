import renderer from 'react-test-renderer';
import { InfoModal } from '../InfoModal';
import { InfoModalProps } from '../../@types/types';
import ReactDOM from 'react-dom';

const props: InfoModalProps = {
  infoModalOpen: true,
  onClose: jest.fn(),
}

describe('InfoModal', () => {
  beforeAll(() => {
    // @ts-ignore
    ReactDOM.createPortal = jest.fn((element, node) => {
      return element
    })
  })

  afterEach(() => {
    // @ts-ignore
    ReactDOM.createPortal.mockClear()
  })

  it('match snapshot', () => {
    const infoModal = renderer.create(<InfoModal {...props} />).toJSON();
    expect(infoModal).toMatchSnapshot();
  });
})