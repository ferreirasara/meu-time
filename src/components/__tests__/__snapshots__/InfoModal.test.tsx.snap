// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoModal match snapshot 1`] = `null`;
